internal/voting/voting.go

```diff
@@ -69,37 +72,40 @@ func dropExtraVoteIDColumn(ctx context.Context, db sql.DB) error {
 	return err
 }
 
-// Approve approves a resolution from a voter.
+// ApproveResolution approves a resolution from a voter.
 // If the resolution does not yet exist, it will be errored,
 // Validators should only vote on existing resolutions.
 // If the voter does not exist, an error will be returned.
 // If the voter has already approved the resolution, no error will be returned.
 // If the resolution has already been processed, no error will be returned.
-func ApproveResolution(ctx context.Context, db sql.TxMaker, resolutionID types.UUID, expiration int64, from []byte) error {
+func ApproveResolution(ctx context.Context, db sql.TxMaker, resolutionID types.UUID, from []byte) error {
 	tx, err := db.BeginTx(ctx)
 	if err != nil {
 		return err
 	}
 	defer tx.Rollback(ctx)
 
-	// Expectation is that the resolution is already created when the voteBody is submitted. and nodes wont submit the voteIDs for events which doesn't have resolutions.
-	userId := types.NewUUIDV5(from)
+	// Expectation is that the resolution is already created when the voteBody
+	// is submitted. Nodes won't submit the voteIDs for events which don't have
+	// resolutions. @charitha: is this still accurate?
```

Comment:
Correct, Votes can only be made for existing resolutions. If resolution doesn't exist, this will fail.